### DISCUSSION TITLE
attemt to index a nil value(local'docStructure')

### DISCUSSION THREAD

**Discussion by SaElmas** [xournalpp/xournalpp#3008] [Q&A] [answered]
For the plugin below, I'm getting the error "attemt to index a nil value(local'docStructure')" docStructure could not being initialized I thing, how can i fix this?
Thanks a lot.
-- Register all Toolbar actions and intialize all UI stuff
function initUi()
  app.registerUi({["menu"] = "SetPaper", ["callback"] = "setpaper", ["accelerator"] = "<Control>l"});
end

local docStructure = app.getDocumentStructure
local numPages = #docStructure["pages"]
local currPage = docStructure["currentPage"]

function setpaper()
    	local result = app.msgbox("Which background do you want to use further?", {[1]="Graph", [2]="Ruled", [3]="Plain"})
	if (result==1) then
		for i=1, numPages do
			app.changeCurrentPageBackground("graph");
		end
   	elseif (result==2) then
		for i=1, numPages do
			app.changeCurrentPageBackground("ruled");
		end
	elseif (result==3) then
		for i=1, numPages do
			app.changeCurrentPageBackground("plain");
		end
    	end
e

**Comment by rolandlo** [comment]
You missed the parentheses in `app.getDocumentStructure()`

**Reply by SaElmas** [reply]
I intentionally erased them since with parenthesis I am getting again an error. Is it possible to lost some of the library functions while uninstalling xournall? It seems that my system does not have app.getDocumentStructure() function.

**Comment by rolandlo** [comment, marked as answer]
Which version of Xournal++ do you use? You must use version 1.1.0+dev to have this function.

**Reply by SaElmas** [reply]
version 1.0.20. It seems that it is my problem. Thank you very much.
